lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+
+	attr_reader :cells
+
+	def initialize
+		self.reset!
+	end
+
+	def cells=(cells)
+		@cells = cells
+	end
+
+	def reset!
+		@cells = [" "," "," "," "," "," "," "," "," "]
+	end
+
+	def display
+		puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+		puts "-----------"
+		puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+		puts "-----------"
+		puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+	end
+
+	def position(cell)
+		self.cells[cell.to_i - 1]
+	end
+
+	def full?
+		self.cells.all? {|cell| cell != " "}
+	end
+
+	def turn_count
+		count = 0
+		self.cells.each {|cell| count+=1 unless cell == " "}
+		count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```